### PR TITLE
Add a tip about using ESNext in template expressions

### PR DESCRIPTION
**Pull request by Justineo**
... to clarify situations discussed in vuejs/vue#6307.

### CODE REVIEW

File: src/v2/guide/syntax.md [hunk 1]
```diff
@@ -80,6 +80,8 @@ These expressions will be evaluated as JavaScript in the data scope of the owner
 
 <p class="tip">Template expressions are sandboxed and only have access to a whitelist of globals such as `Math` and `Date`. You should not attempt to access user defined globals in template expressions.</p>
 
+<p class="tip">ESNext features are available in template expressions with the caveat that the feature set is limited by the **Node.js** version you use.</p>
```
Comment (rpkilby) [inline src/v2/guide/syntax.md]: A minor nitpick, but this supposes that rendering occurs on the server, not on the client.

**Comment by chrisvfritz** [comment]
This isn't really the right place for a note like this, since it's specific to using Babel and `.vue` files. The [vue-template-compiler](https://www.npmjs.com/package/vue-template-compiler) docs would probably be more appropriate, but I think still inadequate, since it's extremely unlikely that anyone will find that note when they encounter an error.

I'm continuing the conversation in the original issue to try to find a better solution. This is definitely unexpected behavior for a user, so I think either a warning when an error occurs, or finding some way to work around the problem is needed.

**Comment by Justineo** [comment]
Thanks Chris. Obviously it's the best if the original issue can be solved.

File: src/v2/guide/syntax.md (hunk 1 above)
Comment (nickmessing) [inline src/v2/guide/syntax.md]: @rpkilby, template validation occurs on the server.

**Comment by nickmessing** [comment]
@Justineo, opened a PR for that (thank you @chrisvfritz for the suggestion). vuejs/vue#6328

**Comment by Justineo** [comment]
Should we still address the difference between precompiled template and runtime template? Like runtime template won't be transformed by Babel or other tools.

**Comment by chrisvfritz** [comment]
@Justineo Good question! My current thinking is that if someone is using SFCs with the runtime-only build, it makes sense for them to expect that the JavaScript they use in a `script` will always work in a `template` as well - so that's the main problem I'm hoping to solve. 

Since Babel is a tool only used in the development environment, rather than bundled with client code, my _hope_ is that it will be intuitive to users that templates parsed on the client won't have access to Babel. Do you think that makes sense?

**Comment by Justineo** [comment]
@chrisvfritz Yes that makes sense.

**Pull request closed without merging**